records/2024-05-23.scalability.short-code-implementation.md

```diff
@@ -0,0 +1,107 @@
+# Short Code Implementation
+
+Date: 2025-05-23
+
+## Status
+
+**DRAFT**.
+
+## Context
+
+Notify was awarded a General Purpose Short Code by the Canadian Telecom Regulators. This ADR goes through when an SMS will be sent through a Short Code or Long Code and the decision behind why a certain method will be chosen.
+
+## Implementation Goals
+1. Assign SMS templates to send through our generic shortcode.
+1. Assign any template to high, normal, or bulk priority automatically (SMS & Email)
+1. Design a way to categorize templates
+1. Override to force a certain template to go through a certain flow
+1. The ability for users to select “Other” and enter their own use case for Template Categories
+
+## Goals to be addressed by another ADR
+1. Figure out dynamic and static pools of numbers
+1. Decide on using AWS Pinpoint or AWS SNS
+Usage of a pool of codes/ dedicated long codes
+
+## Current Context and Assumptions
+
+1. **We assume that ALL Priority SMS Messages will be sent using the Short Code**
+1. For the purposes of this ADR, assume SMSs are sent to a send_sms client call. Under the hood the send_sms call might utilize SMS or Pinpoint.
+1. Currently, we have 3 priority queues - HIGH, MEDIUM, and LOW. Given a template, your message will reach within a certain SLO if sent through a certain queue
+1. A template can be of type PRIORITY, NORMAL, and BULK which ***currently*** maps directly to the HIGH, MEDIUM, and LOW queues. 
```

Comment:
Yeah, that would be nice to change that in the UI and finish the switch to the low/medium/priority terms.